src/torchcodec/_core/SingleStreamDecoder.cpp

```diff
@@ -526,6 +526,12 @@ FrameOutput SingleStreamDecoder::getFrameAtIndexInternal(
   const auto& streamInfo = streamInfos_[activeStreamIndex_];
   const auto& streamMetadata =
       containerMetadata_.allStreamMetadata[activeStreamIndex_];
+
+  std::optional<int64_t> numFrames = getNumFrames(streamMetadata);
+  if (numFrames.has_value()) {
+    // If the frameIndex is negative, we convert it to a positive index
+    frameIndex = frameIndex >= 0 ? frameIndex : frameIndex + numFrames.value();
+  }
```

Comment:
If `numFrames` does _not_ have a value, then I think we need to assert that `frameIndex` is positive, and throw a `std::range_error` if it's not. Otherwise, we'll end up using a negative `frameIndex` further down.